app/Models/Concerns/Transaction/CanBeSorted.php

```diff
@@ -39,4 +40,23 @@ public function scopeSortByType(mixed $query, SortDirection $sortDirection): Bui
         ->selectRaw('transactions.*')
         ->orderBy('transaction_type', $sortDirection->value);
     }
+
+    public function scopeSortByUsername(mixed $query, SortDirection $sortDirection): Builder
+    {
+        return $query->select([
+            'validator_name' => fn ($query) => $query
+                ->selectRaw('wallets.attributes->\'username\'')
+                ->from(function ($query) {
+                    $query
+                        ->selectRaw('case when (NULLIF(LEFT(asset->\'votes\'->>0, 1), \'-\') IS null) then substring(asset->\'votes\'->>0, 2) end as unvote')
+                        ->selectRaw('case when (NULLIF(LEFT(asset->\'votes\'->>0, 1), \'+\') IS null) then substring(asset->\'votes\'->>0, 2) end as vote')
+                        ->selectRaw('case when (NULLIF(LEFT(asset->\'votes\'->>0, 1), \'-\') IS null and asset->\'votes\'->>1 is not null and NULLIF(LEFT(asset->\'votes\'->>1, 1), \'+\') IS null) then substring(asset->\'votes\'->>1, 2) end as votecombination')
+                        ->whereColumn('transactions.id', 'validator_transaction.id')
+                        ->from('transactions', 'validator_transaction');
+                }, 'validator_vote')
+                ->join('wallets', 'wallets.public_key', '=', DB::raw('coalesce(validator_vote.votecombination, validator_vote.vote, validator_vote.unvote)')),
+        ])
+        ->selectRaw('transactions.*')
+        ->orderBy('validator_name', $sortDirection->value);
+    }
```

Comment:
@alexbarnsley 

I think we need to add some tests specifically for this scope. It must be getting called somewhere because we don't have issues with test coverage, but since the query is quite complex, I think it would be good to have some dedicated tests.

Reply:
I've refactored that method so it works correctly now. I've also [added a card](https://app.clickup.com/t/86dvt4djg) to adjust it further to consider known wallets when sorting.

I wrote a more comprehensive set of tests for ascending and descending to have a scenario for each possible use-case:

- Name + Address ASC
- Name + Address ASC
- empty value always last